tensorboard/backend/event_processing/data_provider.py

```diff
@@ -221,51 +227,57 @@ def _list(self, construct_time_series, index):
                 )
         return result
 
-    def _read(self, convert_event, index, downsample):
-        """Helper to read scalar or tensor data from the multiplexer.
+    def _read(
```

Comment:
Hmm, so in my original comment I mostly meant changing the blobs to also use the `_index()` helper.  I'm a little ambivalent about whether the consolidation to have them also use the `_read()` helper is worth it?  It complicates the event converters (since they now need all the extra parameters) and it forces `_read()` to take more parameters and become kind of monolithic, whereas I actually kind of liked how before the `read_*()` implementations could be decomposed into validation, generating the index, and then reading from it.

That said since you've already done the work to consolidate it, totally up to you, just my $0.02.

Reply:
Cool, I’m glad you agree. :-) I also don’t like the smörgåsbord of extra
arguments to… everything. Reverted to just the `_index` simplification.

> That said since you've already done the work to consolidate it,
> totally up to you, just my $0.02.

Not at all—my willingness to rewrite a PR depends only on the value of
the proposed change and not on how many times I’ve already rewritten it.

Reply:
LGTM :)